core/camel-main/src/main/java/org/apache/camel/main/BaseMainSupport.java

```diff
@@ -415,6 +421,15 @@ protected void configurePropertiesService(CamelContext camelContext) throws Exce
         if (op != null) {
             pc.setOverrideProperties(op);
         }
+
+        Optional<String> cloudLocations = pc.resolveProperty(MainConstants.CLOUD_PROPERTIES_LOCATION);
+        if (cloudLocations.isPresent()) {
+            LOG.info("Cloud properties location: {}", cloudLocations);
+            final Properties kp = tryLoadCloudProperties(op, cloudLocations.get());
+            if (kp != null) {
+                pc.setOverrideProperties(kp);
```

Comment:
There is some code above that also set some OverrideProperties - so you would override what may have been added before. So I think you need to check if something exists, and then putAll in that case.